_posts/2020-10-30-snowflake-dataOps.md

```diff
@@ -0,0 +1,154 @@
+```
+create database my_db;
+grant all on database my_db to role pipeline_role;
+
+create schema my_schema;
+grant all on schema my_schema to role pipeline_role;
+```
+
+# Step 4: Update pipeline parameters 
+Last step before creating your pipeline is to update parameters with the naming conventions you used to create your snowflake resources. If you have followed my naming convention you probably will not need to change anything except SnowflakeAccount and SnowflakeSecretsManagerARN.
+Update both parameter files pipeline/aws_seed-cli-parameters.json and aws_seed.json to match resources you created in snowflake
+
+```
+ "SnowflakeAccount": "<ACCOUNTNAME>.<REGION>",
+ "SnowflakeUsername": "pipeline_sys_user",
+ "SnowflakeMigrationDatabaseName": "pipeline_db_migration_plan",
+ "SnowflakeWarehouse": "pipeline_warehouse",
+ "SnowflakeRole": "pipeline_role",
+ "SnowflakeSecretsManagerARN": "<The arn to the secrets manager that holds Snowflake private key>"
+```
+
+```
+  {
+    "ParameterKey": "SnowflakeAccount",
+    "ParameterValue": "<ACCOUNTNAME>.<REGION>"
+  },
+  {
+    "ParameterKey": "SnowflakeUsername",
+    "ParameterValue": "pipeline_sys_user"
+  },
+  {
+    "ParameterKey": "SnowflakeWarehouse",
+    "ParameterValue": "pipeline_warehouse"
+  },
+  {
+    "ParameterKey": "SnowflakeMigrationDatabaseName",
+    "ParameterValue": "pipeline_db_migration_plan"
+  },
+  {
+    "ParameterKey": "SnowflakeRole",
+    "ParameterValue": "pipeline_role"
+  },
+  {
+    "ParameterKey": "SnowflakeSecretsManagerARN",
+    "ParameterValue": "<The arn to the secrets manager that holds Snowflake private key>"
+  }
+```
+
+# Step 4: Deploy your code to AWS
+To deploy the infrastructure for your pipeline, you will need to first setup your aws credentials in your terminal. Once it is done, execute init.sh file.<br/>
+<i>Note: the aws user/role you are running the init script as will need admin-like privileges, e.g. be able to create iam roles</i>
+
+```
+sh pipeline/init.sh
+```
+Make sure all the steps goes through successfully. The above init script executes two major steps:
+1. Creating the cloudformation stack for deploying all the AWS infrastructure including CodePipeline, Codebuild and Codecommit
+2. Pushing your codebase into the newly created Codecommit repository
+
+If the second step fails because of your git config settings, you will need to make sure to run them again
+<center><img src="/img/blog/dataops/init.png" /></center><br/>
+Now you can check codepipeline and see a green end to end deployment
+<left><img src="/img/blog/dataops/codepipeline.png" /></left><br/>
+
+This repo uses Flyway to deploy Snowflake changes. To deploy new changes, all you need is placing your new sql scripts in the sql folder and pushing it to codecommit.
+
+Make sure to follow Flyway the naming conventions. All version sql files should start with V__ and repeatable scripts should start with R___
```

Comment:
Suggestion; maybe add an example here to demonstrate the naming convention and folder structure?

Reply:
I have added a photo example of my files